src/makeTxn.ts

```diff
@@ -262,9 +323,24 @@ export function makeKeyRegistrationTxnWithSuggestedParamsFromObject(
     | 'voteKeyDilution'
     | 'suggestedParams'
     | 'rekeyTo'
-    | 'nonParticipation'
-  >
-) {
+  > & {
+    nonParticipation?: false;
+  }
+): txnBuilder.Transaction;
+export function makeKeyRegistrationTxnWithSuggestedParamsFromObject(
+  o: Pick<
+    RenameProperty<
+      MustHaveSuggestedParams<KeyRegistrationTxn>,
+      'reKeyTo',
+      'rekeyTo'
+    >,
+    'from' | 'note' | 'suggestedParams' | 'rekeyTo'
+  > & {
+    nonParticipation?: true;
```

Comment:
```suggestion
    nonParticipation: true;
```
I am seeing the following code pass incorrectly, but I believe this change should fix it:
```javascript
algosdk.makeKeyRegistrationTxnWithSuggestedParamsFromObject({
    from: account.addr,
    // none of the key params or nonParticipation
    suggestedParams: await client.getTransactionParams().do(),
})
```